apps/web/src/actions/user-profile-mod.ts

```diff
@@ -37,43 +37,58 @@ export const modifyRegistrationData = authenticatedAction
 				personalWebsite,
 				phoneNumber,
 				countryOfResidence,
+				uploadedFile,
 			},
 			ctx: { userId },
 		}) => {
 			const user = await getUser(userId);
 			if (!user) throw new Error("User not found");
-			await Promise.all([
-				db
-					.update(userCommonData)
-					.set({
-						age,
-						gender,
-						race,
-						ethnicity,
-						shirtSize,
-						dietRestrictions: dietaryRestrictions,
-						accommodationNote,
-						phoneNumber,
-						countryOfResidence,
-					})
-					.where(eq(userCommonData.clerkID, user.clerkID)),
-				db
-					.update(userHackerData)
-					.set({
-						isEmailable,
-						university,
-						major,
-						levelOfStudy,
-						schoolID,
-						hackathonsAttended,
-						softwareExperience: softwareBuildingExperience,
-						heardFrom: heardAboutEvent,
-						GitHub: github,
-						LinkedIn: linkedin,
-						PersonalWebsite: personalWebsite,
-					})
-					.where(eq(userHackerData.clerkID, user.clerkID)),
-			]);
+			await db.transaction(async (tx) => {
```

Comment:
I would take these out of a transaction. If one part of the transaction messes up, we do not need to `rollback` the others as well. If you wanted better feedback, you could associate each part with a key and return something like `{success: false, id:section_id}` to highlight the part that failed, but that is not within the scope of this issue.